cp-connector/pkg/controlplane/controlplane_test.go

```diff
@@ -49,8 +69,13 @@ func TestControlPlaneSubscriptionSourceFailsToStart(t *testing.T) {
 	esm := &EventSourceMock{StartFn: func(ctx context.Context, data RegistrationData, ces chan EventUpdate) error {
 		return nil
 	}}
+	um := &fake.UniformInterfaceMock{
+		RegisterIntegrationFn: func(integration models.Integration) (string, error) {
+			return "some-id", nil
+		},
+	}
 	integration := ExampleIntegration{RegistrationDataFn: func() RegistrationData { return RegistrationData{} }}
-	err := New(ssm, esm).Register(context.TODO(), integration)
+	err := New(ssm, esm, um).Register(context.TODO(), integration)
```

Comment:
here we could also check if the integration id was passed to the eventSource and subscriptionSource

Reply:
done